linux/LinuxProcessList.c

```diff
@@ -1504,6 +1543,15 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
 
       LinuxProcessList_recurseProcTree(this, procFd, "task", proc, period);
 
+      if (ss->flags & PROCESS_FLAG_LINUX_CGROUP || hideRunningInContainer) {
+         LinuxProcessList_readCGroupFile(lp, procFd);
+         if (hideRunningInContainer && lp->cgroup &&
+            !(String_startsWith(lp->cgroup, "/user") || String_startsWith(lp->cgroup, "/machine") || String_startsWith(lp->cgroup, "/system")) ) {
+            if (!LinuxProcessList_checkPidNamespace((Process *)lp, procFd))
```

Comment:
Maybe factor the cgroup heuristic into a separate function so it's easier to understand the reason for the checks and tweak them. (Also `(Process *)lp` is just `proc`).

Reply:
Got it.